versioned_docs/version-5.x/configuring-links.md

```diff
@@ -0,0 +1,355 @@
+
+Sometimes we want to parse the param in a specific way. By default, all params are parsed to the screen as strings, but we can define a `parse` property, which will handle parsing, and `stringify` property to convert it back to a string. If we wanted to resolve `user/Wojciech/22` to user's age as a number and a modified id, we could make `Profile`'s config look like this:
+
+```js
+Profile: {
+  path: 'user/:id/:age',
+  parse: {
+    id: id => `user-${id}`,
+    age: Number,
+  },
+  stringify: {
+    id: id => id.replace(/^xxx-/)
+  }
+}
+```
+
+Similarly to web URLs, there is also an option of using query params, which can only be applied to the deepest screen in the URL. To apply it to our config, the URL would look like this: `user?id=Wojciech&age=22` and config would not include params in the `path` property:
+
+```js
+Profile: {
+  path: 'user',
+  parse: {
+    id: id => `user-${id}`,
+    age: Number,
+  },
+  stringify: {
+    id: id => id.replace(/^xxx-/)
+  }
+}
+```
+
+### Specific configuration
+
+Here we will mention some of the consequences of using nested navigators in the config.
+
+1. Using `initialRouteName` makes the screen always appear in the state of a navigator. It makes it impossible to pass params to that screen through the URL unless the screen is explicitly mentioned in it. It is then reasonable for that screen not to take any params or to provide default ones.
+
+2. Having multiple screens of a nested navigator present in the state object is not possible through a one-part URL string. We also cannot provide a URL like (for the above configuration) `/user/home`, because it would be resolved to `HomeStack->Profile->HomeStack->Home`, If we want to get rid of the second `HomeStack`, we can provide an explicit config for the `Home` screen in the first level of config nesting (remember to provide a different path string for each occurrence of a screen in the config):
+
+   ```js
+   config = {
+     HomeStack: {
+       path: 'stack',
+       initialRouteName: 'Profile',
+       screens: {
+         Home: 'home',
+         Profile: 'user',
+       },
+     },
+     Settings: 'settings',
+     Home: 'first',
+   };
+   ```
+
+   Then `/user/first` will resolve to `HomeStack->Profile->Home`.
+
+   We can also make the config not include nested navigator, which will make the URL even longer, but maybe a bit less confusing.
+
+   ```js
+   config = {
+     HomeStack: 'stack',
+     Home: 'home',
+     Profile: 'user',
+     Settings: 'settings',
+   };
+   ```
+
+   Then `/stack/user/home` will resolve to `HomeStack->Profile->Home`.
+
+3. Usage of the nested navigators disables the option of passing any params to the routes in the nested state, except for the last one. It shouldn't be a problem, because only the last part of the nested config should be a screen, while rest being navigators, which don't take any params.
```

Comment:
```suggestion
3. **Usage of the nested navigators disables the option of passing any params to the routes in the nested state, except for the last one.** It shouldn't be a problem, because only the last part of the nested config should be a screen, while rest being navigators, which don't take any params.
```